src/ca65/main.c

```diff
@@ -307,6 +307,14 @@ static void SetSys (const char* Sys)
             NewSymbol ("__OSIC1P__", 1);
             break;
 
+        case TGT_CREATIX_EMUL:
+            NewSymbol ("__CREATIX_EMUL__", 1);
+            break;
+            
+        case TGT_CREATIX_NATIVE:
```

Comment:
Building with emul does not support (ind),x, which is needed for native 
code.

The emul option is needed for code the compiler generates and the existing
libsrc asm files that use (ind),y.

The emul build must waste lots of cylces with emulating every (ind),y
with code that uses (ind),x.

I think the best option would be to switch between emul and native
with an assembler directive.


On 09.05.2015 10:53, Christian Groessler wrote:
>
> In src/ca65/main.c 
> <https://github.com/cc65/cc65/pull/157#discussion_r29991175>:
>
> > @@ -307,6 +307,14 @@ static void SetSys (const char* Sys)
> >              NewSymbol ("__OSIC1P__", 1);
> >              break;
> >
> > +        case TGT_CREATIX_EMUL:
> > +            NewSymbol ("__CREATIX_EMUL__", 1);
> > +            break;
> > +
> > +        case TGT_CREATIX_NATIVE:
>
> Why do you need emul and native libs? You want to have a runtime lib 
> for "creatix", correct? If it uses emul features (and I guess it does, 
> since that was your reason to add it), build it with "emul".
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cc65/cc65/pull/157/files#r29991175>.
>

-- 
Mit freundlichen Grüssen

Rainer Keuchel